api/test/integration/env/configurations/vulnerability/manager/healthcheck/healthcheck.py

```diff
@@ -64,16 +64,22 @@ def set_healthcheck_state(state):
         # Read state from file instead of api request.
         agents_node = get_healthcheck_state()
 
-    checks_list = list()
+    # Create a checks list with the manager health base at first
+    checks_list = [get_manager_health_base()]
+
     expected_log = "grep -q 'wazuh-modulesd:vulnerability-detector: INFO: (5471): Finished vulnerability assessment " \
         "for agent '\\''{}'\\''' /var/ossec/logs/ossec.log"
 
-    if socket.gethostname() == 'wazuh-master':
-        checks_list.append(get_manager_health_base())
-
     for agent in agents_to_check:
         if agents_node[agent] == socket.gethostname():
-            # Append only logs that are expected in the current node.
+            # Append only logs that are expected in the current node to the checks list
             checks_list.append(check(system(expected_log.format(agent))))
 
+    nodes_to_check = list(dict.fromkeys(list(agents_node.values())))
+    for node in nodes_to_check:
+        if node == socket.gethostname():
```

Comment:
Do we really need to iterate over all the nodes? This could be reduced to:

```python
if socket.gethostname() in nodes_to_check:
```

Reply:
Loop deleted to use the line suggested.

Test results:

```
test_rbac_black_vulnerability_endpoints.tavern.yaml 
	 1 passed, 3 warnings

test_rbac_white_vulnerability_endpoints.tavern.yaml 
	 1 passed, 3 warnings

test_vulnerability_endpoints.tavern.yaml 
	 1 passed, 3 warnings
```